### DISCUSSION TITLE
text input : from params to body

### DISCUSSION THREAD

**Discussion by amorin-gladia** [fastapi/fastapi#5200] [Questions] [answered]
Hi every one,

My api is using text in input with post method.
Right now, the text is a parameter of the request like "/?text=Input%20text%20to%20perform"
But as I can have huge text, I would like to pass all text parameters to the body of the request.

Right now, I can't figure out how to do this. Is there someone who know how ?

**Comment by JarroVGIT** [comment, marked as answer]
This is best achieved by using `Body()` as default value for your parameter. Please read the docs [here](https://fastapi.tiangolo.com/tutorial/body/). It is pretty well described.

Below is a fully working example and should run as is:
```python
from fastapi import FastAPI, Body

app = FastAPI()

@app.post("/")
async def root(text: str = Body(), other_text: str = Body()):
    return {"response_text":text, "response_other_text": other_text}


if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8000)
```

Result using curl:
```bash
(.v3.8.13) jarro@MBP-van-Jarro fastapi-github-issues % curl -X 'POST' \
  'http://0.0.0.0:8000/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "text": "this is a very long sentence!",
  "other_text": "This is also a very long sentence!"
}'
{"response_text":"this is a very long sentence!","response_other_text":"This is also a very long sentence!"}%  
```

**Comment by amorin-gladia** [comment]
Thank very much !
